epictrack-web/src/components/myWorkplans/MyWorkplanContainer.tsx

```diff
@@ -1,32 +1,43 @@
-import { useContext } from "react";
+import { useContext, useRef } from "react";
 import { ETPageContainer } from "../shared";
 import { MyWorkplansContext } from "./MyWorkPlanContext";
 import CardList from "./CardList";
 import { Grid } from "@mui/material";
 import Filters from "./Filters";
 import { AssigneeToggle } from "./Filters/AssigneeToggle";
+import { useOnScreen } from "../../hooks";
 
 const WorkPlanContainer = () => {
-  const {} = useContext(MyWorkplansContext);
+  const { loadMoreWorkplans, loadingWorkplans } =
+    useContext(MyWorkplansContext);
+  const ref = useRef<HTMLDivElement>(null);
+  const isVisible = useOnScreen(ref);
+
+  if (isVisible && !loadingWorkplans) {
+    setTimeout(() => loadMoreWorkplans(), 250);
```

Comment:
Can we create a separate component for the trigger component. And logic like this should be wrapped up in something, a useEffect for instance. Also "setTimeout(() => loadMoreWorkplans(), 250);" let's try doing [throttling ](https://www.geeksforgeeks.org/javascript-throttling/)instead of this.

Reply:
We can import throttle from redash like we do with debouncing sometimes:

![image](https://github.com/bcgov/EPIC.track/assets/91914654/adacc60e-0dbe-4a19-bb3f-7ad6629b0fa8)
and we can make the throttle interval like 2 seconds